### PR TITLE
[IndexTable] Change scroll bar container z-index so that it is under the Card shadow bevel

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris-internal/issues/1341

### WHAT is this pull request doing?

Before, the `IndexTable` scrollbar container was z-index `35` which was under the `ShadowBevel` of `LegacyCard` (z-index `101`) but on top of the `ShadowBevel` on `Card` (z-index `32`). This PR lowers the scrollbar container to z-index `30`. This bug was not caught before as all our `IndexTable` stories were using `LegacyCard`.

|Before|After|
|-|-|
|Screenshot 2024-01-08 at 1 24 13 PM|Screenshot 2024-01-08 at 1 21 54 PM|
|Screenshot 2024-01-08 at 12 49 54 PM|Screenshot 2024-01-08 at 12 48 13 PM|

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)

### CODE REVIEW

File: .changeset/green-owls-study.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Changed IndexTable scroll bar container z-index so that it is under the Card shadow bevel
```
Comment (chloerice) [inline .changeset/green-owls-study.md]:
```suggestion
Changed `IndexTable` scroll bar container z-index so that it is under the `Card` shadow bevel
```

**Review by chloerice — APPROVED** [review]
💯 🙌🏽 🚀

**Review by lgriffee — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
💯 🌟

**Pull request merged**